### DISCUSSION TITLE
MySQL Connection refused when using parallel of Gitlab job with paratest

### DISCUSSION THREAD

**Discussion by kayw-geek** [paratestphp/paratest#812] [Q&A]
We have many test case in our project, so we split it into multi-testsuite.

Before that, we using [parallel of Gitlab job](https://docs.gitlab.com/ee/ci/yaml/#parallel) to improve test speed, now I want to using the paratest with parallel of Gitlab job to improve it, but when I used it, I encountered some problems as following.

`mysqli_sql_exception: Connection refused`

`./vendor/bin/paratest --processes=2 --passthru-php="'-d' 'pcov.enabled=1'" --coverage-php="xxx.php-coverage" --log-junit coverage-report/report.xml --coverage-cobertura="coverage-report/xxx.xml" --testdox --verbose --colors=always -c tests/xxx.xml`

Is there any solution to this problem?

**Comment by Slamdunk** [comment]
It doesn't appear to be related to this library, which by default has no relationship with external resources such MySQL.
Be sure to use a different database for each process: https://github.com/paratestphp/paratest#test-token

**Comment by kayw-geek** [comment]
We have 23 concurrent jobs and each job have 2 processes running test, Is it possible that the cause of this problem is too many tests concurrent execution leads to frequent creation and closing of databases.
